spec/services/event_log/journey_executor_spec.rb

```diff
@@ -1,65 +0,0 @@
-# frozen_string_literal: true
```

Comment:
shouldn't these tests be reproduced in some form in the new generic event world?

Reply:
Short answer, no. Long answer... So from the perspective of the runner we only know about whether an eventable is valid/invalid by an integration. We test the handler in the eventable model spec. We test the trigger in the STI model spec. Should we test the integration here at the unit level? I don't think so. Should we test these specific transitions in a workflow somewhere: if there's business value in doing so, yeah. But still not here. Happy to add a task to add some workflow/acceptance tests that cover these integrated behaviours. I'm not convinced we'd be learning/avoiding a breaking change in the behaviour by adding them back in, though. We also do a full integration at the request level for all events that are currently possible to create and the only thing that varies in different transitions is whether the eventable is valid.

Reply:
Presumably whether the transitions are valid themselves are tested inside of the state machine specs, too.

Reply:
I guess you are saying these are integration tests not unit tests, so shouldn't be here?

my concern is that simply deleting these tests and relying purely on unit tests does not offer guarantees that all the expected behaviours will continue to work. 

For example, where there are multiple events for an `eventable`, these should be ordered by timestamp and applied in that order: this may or may not result in a change to the `eventable`. 

I don't think simply testing each event in isolation does not replicate the real-life sceanario with multiple events?

Reply:
So I guess I'm saying is if you could clarify a specific behaviour that is missed without an integration test then I'll add it separately.

Reply:
Calling the trigger method in the applied order is tested, yeah.

Reply:
E.g.:
System has a redirect to A with a timestamp at 15:00
Some time later a new event arrives: redirect to B with timestamp at 14:00
=> net result is log the event but don't change the destination

Reply:
Okay, PR and story incoming :)

Reply:
that is just one example ^ 
... ideally, I think all the existing tests should persist

Reply:
https://dsdmoj.atlassian.net/browse/P4-2322

Reply:
I am concerned that the generic events runner (already merged) doesn't have tests of complex cases (such as 2 unordered redirects) and I'd like to see one or more test(s) for complex cases...